superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -105,7 +106,24 @@ class VisualizeModal extends React.PureComponent {
       sql: this.props.query.sql,
       dbId: this.props.query.dbId,
     };
-    window.open('/superset/sqllab_viz/?data=' + JSON.stringify(vizOptions));
+    $.ajax({
+      type: 'POST',
+      url: '/superset/sqllab_viz/',
+      async: false,
+      data: {
+        data: JSON.stringify(vizOptions),
+      },
+      success: (url) => {
+        window.open(url);
+      },
+      error: (error) => {
```

Comment:
since this is redirecting to explore page, adding alert in sqllab won't be intuitive, I'll delete this error handling.